spirit/comment/forms.py

```diff
@@ -150,13 +151,26 @@ def __init__(self, user=None, *args, **kwargs):
 
     def clean_file(self):
         file = self.cleaned_data['file']
+        file_mime = magic.from_buffer(file.read(), mime=True)
```

Comment:
`.read()` will load the whole file in memory. The magic example does a `.read(1024)`, maybe we should do that and increase it when someone found it to short for some file format.

Reply:
Oh, the `from_buffer` may raise a `magic.MagicException`, so better wrap it in a try..except and raise a `ValidationError(_("Unsupported file"))`.

Reply:
So it appears that `.read(1024)` gives me issues with `.doc` and `.docx` files (`.pdf` works fine). Even with a 9 kb `.doc` file, it will raise the MIME validation error and strangely enough, it thinks the MIME type is something completely different like `application/zip` instead of `application/msword`.

These issues however don't come up at all if I raise it to `.read(8096)`.

Reply:
That's bad news. I think we can check `if isinstance(file, TemporaryUploadedFile)` [0] and call magic `from_file(file.temporary_file_path())` or else just `from_buffer(file.read())` (since the file content is in memory anyway). I can fix it later for you.

[0] https://docs.djangoproject.com/en/1.11/ref/files/uploads/#django.core.files.uploadedfile.TemporaryUploadedFile